app/Services/UserService.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Services;
+
+use App\Http\Requests\Api\Auth\LoginRequest;
+use App\Http\Requests\Api\Auth\RegisterRequest;
+use App\Http\Resources\ErrorResource;
+use App\Http\Resources\SuccessResource;
+use App\Models\User;
+use App\Repositories\OfferRepository;
+use App\Repositories\StatusRepository;
+use App\Repositories\UserRepository;
+use Exception;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Hash;
+
+class UserService
+{
+    protected $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function login(LoginRequest $request)
+    {
+        $user = $this->userRepository->findWhere(['email' => $request->input('email')])->first();
+
+        if (!$user || !$this->checkPassword($request->input('password'), $user->password)) {
+            return response([
+                'message' => ['These credentials do not match our records.']
+            ], 404);
+        }
+
+        try {
+            $token = $user->createToken('authToken')->plainTextToken;
+            return new SuccessResource(Response::HTTP_CREATED, "Logged In successfully.", [
+                'user' => $user,
+                'token' => $token
+            ]);
+        } catch (Exception $e) {
+            return new ErrorResource(Response::HTTP_INTERNAL_SERVER_ERROR, "Internal Server Error", ["An unexpected error occured."]);
+        }
+    }
+
+    public function logout(Request $request)
+    {
+        try {
+            $request->user()->currentAccessToken()->delete();
+            return new SuccessResource(Response::HTTP_CREATED, "Logged out successfully.");
+        } catch (Exception $e) {
+            return new ErrorResource(Response::HTTP_INTERNAL_SERVER_ERROR, "Internal Server Error", ["An unexpected error occured."]);
+        }
+    }
+
+    public function register(RegisterRequest $request)
+    {
+        try {
+            $data = $request->validated();
+
+            if( $request->hasFile('image')) {
+                $imagePath = $request->file('image')->store('profile_pictures','public');
+                $data = array_merge($request->validated(), ['image' => $imagePath]);
+            }
+
+            $user = $this->userRepository->create($data);
+
+            $token = $user->createToken('authToken')->plainTextToken;
+            return new SuccessResource(Response::HTTP_CREATED, "Registred successfully.", [
+                'user' => $user,
+                'token' => $token
+            ]);
+
+        } catch (Exception $e) {
+            return new ErrorResource(Response::HTTP_INTERNAL_SERVER_ERROR, "Internal Server Error", ["An unexpected error occured."]);
```

Comment:
@aminetiyal please note that the last attribute ($errors) is optional and you don't need to pass it if there is no need for that

Reply:
[commit](https://github.com/BuildForSDGCohort2/Team-248-Backend/pull/29/commits/32624509bb02bddc8f6c1723888c63536335e239)